irony-cdb.el

```diff
@@ -114,6 +128,57 @@ for files that it cannot handle."
 ;; Functions
 ;;
 
+(defun irony-cdb--choose-closest-path (file paths)
+  "Find the \"best\" path in PATHS matching FILE
+
+If any paths in PATHS is belongs to the same directory
+or a subdirectory of file, we disregard other candidates.
+
+For remaining candidates, \"nearest\" is measured as abs. difference
+in path depth.
+- We prefer deeper paths at level +N to those at level -N.
+- If multiple paths are equally good, we return the last one.
+
+Returns nil if paths isn't a list of at least one element.
+"
+  (when (listp paths)
+    (let ((paths (or
+                  ;; if we find a cdb in cwd or below, don't consider other candidates
```

Comment:
> which lives in or under that directory will match, by definition. I believe the children-first test (based on the test you provided) shows this works.

Yes you are right, I just don't get it for some reasons.



> That's exactly what the first version I wrote did and I threw that one away because (I thought) we wanted:
> 
> /foo/bar/build/cdb (found by "build/", one level up)
> to win over 
> /cdb (found by ".", two levels up)
> 
> ...when the search begins at /foo/bar/x.cpp. See #237 (comment).
> 
> Git reflog still had it - is this what you want?

No I like the current behavior, it is what I want. I just don't get all of the logic but it's not big deal.